src/Sylius/Bundle/LocaleBundle/Checker/LocaleUsageChecker.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\LocaleBundle\Checker;
+
+use Doctrine\ORM\EntityManagerInterface;
+use Sylius\Component\Locale\Context\LocaleNotFoundException;
+use Sylius\Component\Resource\Metadata\RegistryInterface;
+use Sylius\Component\Resource\Repository\RepositoryInterface;
+
+final class LocaleUsageChecker implements LocaleUsageCheckerInterface
+{
+    public function __construct(
+        private RepositoryInterface $localeRepository,
+        private RegistryInterface $resourceRegistry,
+        private EntityManagerInterface $entityManager,
+    ) {
+    }
+
+    /**
+     * @throws LocaleNotFoundException
+     */
+    public function isUsed(string $localeCode): bool
+    {
+        $locale = $this->localeRepository->findOneBy(['code' => $localeCode]);
+
+        if (null === $locale) {
+            throw new LocaleNotFoundException();
+        }
```

Comment:
We have the existence check both here and [here](https://github.com/Sylius/Sylius/pull/15026/files#diff-4daac477cdf64e68b4f917f0f544817e11f4a6d514f7c265f174bf2924be9c2aR99-R103). I suppose we could only leave it here and catch an exception in the command

Reply:
The check in the command is used to make sure we call the `getCode` method on an object, not null.

The check in the `LocaleUsageChecker` exists because somebody might want to use this service by just passing a locale code.

As I do `remove`, I need a `Locale` entity from the repository anyway, so I need to check against null anyway, to make sure it's never null.